packages/typespec-test/test/openai_modular/generated/typespec-ts/src/models/models.ts

```diff
@@ -610,3 +610,23 @@ export type ImageSize = string;
 /** The format in which the generated images are returned. */
 /** "url", "b64_json" */
 export type ImageGenerationResponseFormat = string;
+
+/** Optional settings to control how fields are processed when using a configured Azure Cognitive Search resource. */
+export interface AzureCognitiveSearchIndexFieldMappingOptions {
+  /** The name of the index field to use as a title. */
+  titleField?: string;
+  /** The name of the index field to use as a URL. */
+  urlField?: string;
+  /** The name of the index field to use as a filepath. */
+  filepathField?: string;
+  /** The names of index fields that should be treated as content. */
+  contentFieldNames?: string[];
+  /** The separator pattern that content fields should use. */
+  contentFieldSeparator?: string;
+  /** The names of fields that represent vector data. */
+  vectorFields?: string[];
+}
+
+/** The type of Azure Cognitive Search retrieval query that should be executed when using it as an Azure OpenAI chat extension. */
+/** "simple", "semantic", "vector", "vectorSimpleHybrid", "vectorSemanticHybrid" */
+export type AzureCognitiveSearchQueryType = string;
```

Comment:
Looks a little strange, Are we generating string for extensible enum now ?
```
enum AzureCognitiveSearchQueryType {
  @doc("Represents the default, simple query parser.")
  simple: "simple",

  @doc("Represents the semantic query parser for advanced semantic modeling.")
  semantic: "semantic",

  @doc("Represents vector search over computed data.")
  vector: "vector",

  @doc("Represents a combination of the simple query strategy with vector data.")
  vectorSimpleHybrid: "vectorSimpleHybrid",

  @doc("Represents a combination of semantic search and vector data querying.")
  vectorSemanticHybrid: "vectorSemanticHybrid",
}
```

Reply:
We always generate string for extensible enum. I remember you have an issue to enrich the knownable values for extensible enum?